internal/plugin/destroyer.go

```diff
@@ -66,12 +66,18 @@ func (c *destroyerClient) destroy(
 	ctx context.Context,
 	args funcspec.Args,
 	internal *pluginargs.Internal,
+	declaredResourcesResp *component.DeclaredResourcesResp,
+	destroyedResourcesResp *component.DestroyedResourcesResp,
 ) error {
 	// Run the cleanup
 	defer internal.Cleanup.Close()
 
 	// Call our function
-	_, err := c.Client.Destroy(ctx, &pb.FuncSpec_Args{Args: args})
+	resp, err := c.Client.Destroy(ctx, &pb.FuncSpec_Args{Args: args})
+
+	declaredResourcesResp.DeclaredResources = resp.DeclaredResources.Resources
+	destroyedResourcesResp.DestroyedResources = resp.DestroyedResources.DestroyedResources
```

Comment:
We should probably check this error above? Previously nothing was returned (`_, err`) so we could just return that through, but now we actually get a response, as well as try to read the responses contents which I _think_ could be `nil` in which case we'd get nil pointer dereferences? Looking at the `callDynamicFunc2()` invocation below, which I _think_ is related to this, we could possibly return `nil, err`, is that correct?

Reply:
Good point! We should check for an error on L77 and return it. That way the declared/destroyed resources of the response aren't referenced, leading to a nil pointer dereference.